parquet/variants/primitive.go

```diff
@@ -0,0 +1,674 @@
+	case primitiveNull:
+		return "Null"
+	case primitiveFalse, primitiveTrue:
+		return "Boolean"
+	case primitiveInt8:
+		return "Int8"
+	case primitiveInt16:
+		return "Int16"
+	case primitiveInt32:
+		return "Int32"
+	case primitiveInt64:
+		return "Int64"
+	case primitiveDouble:
+		return "Double"
+	case primitiveDecimal4:
+		return "Decimal4"
+	case primitiveDecimal8:
+		return "Decimal8"
+	case primitiveDecimal16:
+		return "Decimal16"
+	case primitiveDate:
+		return "Date"
+	case primitiveTimestampMicros:
+		return "Timestamp(micros)"
+	case primitiveTimestampNTZMicros:
+		return "TimestampNTZ(micros)"
+	case primitiveFloat:
+		return "Float"
+	case primitiveBinary:
+		return "Binary"
+	case primitiveString:
+		return "String"
+	case primitiveTimeNTZ:
+		return "TimeNTZ"
+	case primitiveTimestampNanos:
+		return "Timestamp(nanos)"
+	case primitiveTimestampNTZNanos:
+		return "TimestampNTZ(nanos)"
+	case primitiveUUID:
+		return "UUID"
+	}
+	return "Invalid"
+}
+
+func validPrimitiveValue(prim primitiveType) error {
+	if prim < primitiveNull || prim > primitiveUUID {
+		return fmt.Errorf("invalid primitive type: %d", prim)
+	}
+	return nil
+}
+
+func primitiveFromHeader(hdr byte) (primitiveType, error) {
+	// Special case the basic type of Short String and call it a Primitive String.
+	bt := BasicTypeFromHeader(hdr)
+	if bt == BasicShortString {
+		return primitiveString, nil
+	} else if bt == BasicPrimitive {
+		prim := primitiveType(hdr >> 2)
+		if err := validPrimitiveValue(prim); err != nil {
+			return primitiveInvalid, err
+		}
+		return prim, nil
+	}
+	return primitiveInvalid, fmt.Errorf("header is not of a primitive or short string basic type: %s", bt)
+}
+
+func primitiveHeader(prim primitiveType) (byte, error) {
+	if err := validPrimitiveValue(prim); err != nil {
+		return 0, err
+	}
+	hdr := byte(prim << 2)
+	hdr |= byte(BasicPrimitive)
+	return hdr, nil
+}
+
+// marshalPrimitive takes in a primitive value, asserts its type, then marshals the data according to the Variant spec
+// into the provided writer, returning the number of bytes written.
+//
+// Time can be provided in various ways- either by a time.Time struct, or by an int64 when the EncodeAs{Date,Time,Timestamp}
+// options are provided. By default, timestamps are written as microseconds- to use nanoseconds pass in EncodeTimeAsNanos.
+// Timezone information can be determined from a time.Time struct. Otherwise, by default, timestamps will be written with
+// local timezone set.
+func marshalPrimitive(v any, w io.Writer, opts ...MarshalOpts) (int, error) {
+	var allOpts MarshalOpts
+	for _, o := range opts {
+		allOpts |= o
+	}
+	switch val := v.(type) {
+	case bool:
+		return marshalBoolean(val, w), nil
+	case int:
+		return marshalInt(int64(val), w), nil
+	case int8:
+		return marshalInt(int64(val), w), nil
+	case int16:
+		return marshalInt(int64(val), w), nil
+	case int32:
+		return marshalInt(int64(val), w), nil
+	case int64:
+		if allOpts&MarshalAsTime != 0 {
```

Comment:
instead of needing this option, why not just use `arrow.Time32` or `arrow.Time64`? if they want a Time they should pass that.